app/login/page.tsx

```diff
@@ -0,0 +1,348 @@
+'use client'
+
+import './login.styles.css'
+
+import React, { useMemo, useState } from 'react'
+import { useRouter } from 'next/navigation'
+import { ArrowRight, ArrowUpRight, Dot, Loader2, Pencil } from 'lucide-react'
+
+interface ErrorsProps {
+  workEmail?: string
+}
+
+interface Tenant {
+  name: string
+  region: { name: string; dns: string }
+  state: string
+}
+
+enum TenantState {
+  HEALTHY = 'HEALTHY',
+  EXPIRED = 'EXPIRED',
+}
+
+const BASE_URL = 'https://api.signoz.cloud/v2'
+
+const trustBadges = [
+  {
+    icon: '/svgs/icons/hipaa.svg',
+    text: 'HIPPA',
+  },
+  {
+    icon: '/svgs/icons/SOC-2.svg',
+    text: 'SOC2',
+  },
+  {
+    text: 'Privacy',
+    url: 'https://www.signoz.io/privacy',
+  },
+  {
+    text: 'Security',
+    url: 'https://www.signoz.io/security',
+  },
+]
+
+export default function Login() {
+  const [workEmail, setWorkEmail] = useState('')
+
+  const [errors, setErrors] = useState<ErrorsProps>({})
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [submitSuccess, setSubmitSuccess] = useState(false)
+  const [submitFailed, setSubmitFailed] = useState(false)
+  const [userTenants, setUserTenants] = useState([])
+
+  const [noDeployments, setNoDeployments] = useState(false)
+  const router = useRouter()
+
+  const handleEmailUpdate = (event) => {
+    const { value } = event.target
+    setWorkEmail(value)
+  }
+
+  const validateForm = () => {
+    let errors = {}
+
+    if (!workEmail.trim()) {
+      errors['workEmail'] = 'Work email is required'
+    } else if (!isValidCompanyEmail(workEmail)) {
+      errors['workEmail'] = 'Please enter a valid company email'
+    }
+
+    setErrors(errors)
+
+    return Object.keys(errors).length === 0
+  }
+
+  const isValidEmail = (email) => {
+    // Basic email validation regex
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
+    return emailRegex.test(email)
+  }
+
+  function isValidCompanyEmail(email) {
+    // Regular expression pattern to match valid company email domains
+    var companyEmailPattern =
+      /@(?!gmail|yahoo|hotmail|outlook|live|icloud)[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/
+
+    // Check if the email matches the email format and the company email pattern
+    return isValidEmail(email) && companyEmailPattern.test(email)
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+
+    setSubmitFailed(false)
+
+    const isFormValid = validateForm()
+
+    if (isFormValid) {
+      handleLogin()
+    }
+  }
+
+  const handleError = () => {
+    setSubmitFailed(true)
+    window.scrollTo({
+      top: 100,
+      behavior: 'smooth',
+    })
+  }
+
+  const handleNoDeployments = () => {
+    setNoDeployments(true)
+  }
+
+  const handleLogin = async () => {
+    setIsSubmitting(true)
+    setSubmitFailed(false)
+
+    const verifyWorkSpaceSetupURL = `${BASE_URL}/deployments/esearch?q=${workEmail}`
```

Comment:
All query-parameters should be url encoded @YounixM.

Reply:
Done ✅